### PR TITLE
Use go-getter with secrets as well

### PR DESCRIPTION
**Pull request by itscaro**
Before this PR, only `values` can be used with go-getter and not `secrets`. The PR implement `remote FS` for secrets as well to maintain the consistency.

```
#{{ $configRepo := "git::https://example.xxx/yyy/zzz.git" }}

environments:
  '{{ .Environment.Name }}':
    values:
    #{{- if .Environment.Values }}
    - {{- .Environment.Values | toYaml | nindent 6 }}
    #{{- else}}
    - ../envs/_default.yaml
    - ../envs/{{ $env }}.yaml
    - '{{ $configRepo }}@/{{ $env }}.yaml?ref={{ env "CONFIG_REPO_REF" | default "main" }}'
    #{{- end }}
    secrets:
    #{{- not if .Environment.Values }}
    - ../envs/{{ $env }}.secrets.yaml
    - '{{ $configRepo }}@/{{ $env }}.secrets.yaml?ref={{ env "CONFIG_REPO_REF" | default "main" }}'
    #{{- end }}
```

Local test:
we can see: `skipping missing values file matching` and `skipping missing secrets file matching`
```
C:\Users\qtran\workspace\github\helmfile\dist\helmfile-dev.exe -f input.yaml template --debug 2>&1 | grep helmfile/helmfile
13:   - git::https://github.com/helmfile/helmfile.git@test/e2e/template/helmfile/testdata/snapshot/pr_560/values.yaml?ref=main
15:   - git::https://github.com/helmfile/helmfile.git@test/e2e/template/helmfile/testdata/snapshot/pr_560/secrets.yaml?ref=main
13:   - git::https://github.com/helmfile/helmfile.git@test/e2e/template/helmfile/testdata/snapshot/pr_560/values.yaml?ref=main
15:   - git::https://github.com/helmfile/helmfile.git@test/e2e/template/helmfile/testdata/snapshot/pr_560/secrets.yaml?ref=main
dir: /helmfile/helmfile.git
downloading git::https://github.com/helmfile/helmfile.git?ref=main to values\https_github_com_helmfile_helmfile_git.ref=main
client: {Ctx:context.Background Src:git::https://github.com/helmfile/helmfile.git?ref=main Dst:C:\Users\qtran\AppData\Local\helmfile\values\https_github_com_helmfile_helmfile_git.ref=main
skipping missing values file matching "git::https://github.com/helmfile/helmfile.git@test/e2e/template/helmfile/testdata/snapshot/pr_560/values.yaml?ref=main"
dir: /helmfile/helmfile.git
skipping missing secrets file matching "git::https://github.com/helmfile/helmfile.git@test/e2e/template/helmfile/testdata/snapshot/pr_560/secrets.yaml?ref=main"
```

When the file exist we can see in the log
```
Decrypting secret C:\Users\xxx\AppData\Local\helmfile\values\https_xxx_app-deploy-config_git.ref=main\xxx\_default.secrets.yaml
```

### CODE REVIEW

**Comment by yxxhero** [comment]
@itscaro adding some unit tests will be better.

**Comment by itscaro** [comment]
@yxxhero yes just PR to discuss about the feature first :)

**Comment by yxxhero** [comment]
@itscaro yeah. could you show an example?

**Comment by itscaro** [comment]
@yxxhero just added an example in description, still testing on local :)

**Comment by itscaro** [comment]
@yxxhero Humh any idea how we test this part, I found no existing test for go getter with either values or secrets

**Comment by yxxhero** [comment]
@itscaro I will see how to add a unit test. you can test it locally, and copy the process to the description.

**Comment by itscaro** [comment]
@yxxhero I was trying to use e2e test to check for output, but there are random temporary directories in output, any idea to just check if output contains some string? I've not looked how snapshot_test.go works thoroughly.

**Comment by mumoshu** [comment]
@itscaro I think you can specify HELMFILE_TEMPDIR to make temp values file paths a bit more predictable and regex-replace the paths in the real output to make it stable enough for comparison with the snapshot.

https://github.com/helmfile/helmfile/blob/f7b3fb4432dba7889884a1a741626b5958496fa6/pkg/envvar/const.go#L7

**Comment by itscaro** [comment]
@mumoshu thanks, I have thought of it too, not have time to test yet.

File: pkg/state/create.go [hunk 1]
```diff
@@ -293,9 +293,15 @@ func (c *StateCreator) scatterGatherEnvSecretFiles(st *HelmState, envSecretFiles
 		},
 		func(id int) {
 			for secret := range secrets {
+				urlOrPath := secret.path
+				localPath, err := c.remote.Locate(urlOrPath)
```
Comment (yxxhero) [inline pkg/state/create.go:297]: if err != nil
then?

**Comment by yxxhero** [comment]
need to update docs.

**Comment by itscaro** [comment]
> > This logger depends on how fast the output is capture we have one line log ou two-line log so the log prefix helm> differs :( any idea?
> 
> Is that due to that Helmfile runs multiple Helm commands concurrently? Then I can recommend setting `concurrency` to 1. It defaults to the number of releases in the helmfile.yaml so it's fully concurrent by default. Explicitly setting it to 1 completely disables the concurrency. We do the same in unit tests like `apply_test.go`.

with --enable-live-output, it seems the output is idempotent 


@yxxhero  will update doc

File: pkg/state/create.go (hunk 1 above)
Comment (itscaro) [reply pkg/state/create.go:297]: then urlOrPath := secret.path

the code is copied as-is from `values` process

**Comment by itscaro** [comment]
- [x] update doc
- [x] add `remote>` prefix to logs in `remote.go`
- [x] use `--concurrency=1`

**Review by yxxhero — APPROVED** [review]
LGTM

**Pull request merged**